utils/dataconvert/dataconvert.cpp

```diff
@@ -1974,6 +2130,116 @@ int64_t DataConvert::convertColumnTimestamp(const char* dataOrg, CalpontDateTime
   return value;
 }
 
+
+int64_t DataConvert::convertArrowColumnTime32(int32_t timeVal)
+{
+  int64_t value = 0;
+  // convert millisecond to time
+  int inHour, inMinute, inSecond, inMicrosecond;
+  inHour = inMinute = inSecond = inMicrosecond = 0;
+  bool isNeg = false;
+  if (timeVal < 0)
+    isNeg = true;
+  inHour = timeVal / 3600000;
+  // inHour %= 24;
+  inMinute = (timeVal - inHour * 3600000) / 60000;
+  // inMinute %= 60;
+  inSecond = (timeVal - inHour * 3600000 - inMinute * 60000) / 1000;
+  // inSecond %= 60;
+  inMicrosecond = (timeVal - inHour * 3600000 - inMinute * 60000 - inSecond * 1000) * 1000;
+  if (isTimeValid(inHour, inMinute, inSecond, inMicrosecond))
+  {
+    Time atime;
+    atime.hour = inHour;
+    atime.minute = inMinute;
+    atime.second = inSecond;
+    atime.msecond = inMicrosecond;
+    atime.is_neg = isNeg;
+
+    memcpy(&value, &atime, 8);
+  }
+  else
+  {
+    // Emulate MariaDB's time saturation
+    if (inHour > 838)
+    {
+      Time atime;
+      atime.hour = 838;
+      atime.minute = 59;
+      atime.second = 59;
+      atime.msecond = 999999;
+      atime.is_neg = false;
+      memcpy(&value, &atime, 8);
+    }
+    else if (inHour < -838)
+    {
+      Time atime;
+      atime.hour = -838;
+      atime.minute = 59;
+      atime.second = 59;
+      atime.msecond = 999999;
+      atime.is_neg = false;
+      memcpy(&value, &atime, 8);
+    }
```

Comment:
This code part is similar to `DataConvert::convertColumnTime` which also has no `else {}` block. Comments in `DataConvert::convertColumnTime` shows if `if{}` and `else if{}` don't match, return 0 then.

Reply:
`DataConvert::convertColumnTime` is setting `status = -1` in the outer `else` block. Please fix this here as well as in `DataConvert::convertArrowColumnTime64` so that the caller of these functions handles a non-zero return code, as you are handling the non-zero return code for `dataconvert::DataConvert::convertArrowColumnDatetime`.